### DISCUSSION TITLE
Serial communication external multi-protocol (4-in-1) RF module

### DISCUSSION THREAD

**Discussion by dgehriger** [OpenI6X/opentx#393] [General]
Hi,

I'm successfully using OpenI6X with the external iRangeX IRX4 Plus RF module, using PPM communication. The firmware for the external module (https://github.com/pascallanger/DIY-Multiprotocol-TX-Module) explicitly mentions that it is possible to also use **serial** communication with remotes running OpenTX.

I believe this has not been ported to OpenI6X yet, and I would be willing to implement it. To start, I need to know the current status, what pins to use and if any similar project had been attempted.

Can anyone comment on the following:

1. Serial comm with the external "[DIY Multiprotocol TX Module](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module)" is not yet implemented, correct?
2. Issue #120 mentions that "This implementation is possible, serial driver examples are ready in aux serial driver and crsf confiruation. This feature need developer with 4in1 module to sync source with current OpenTX master and test it.", but I don't fully understand what the author of this post wants to say. Would it be possible to elaborate?
3. What serial pin to use? Should the _AUX Serial_ port (PA9) be used for this, or any other recommendation?

Thank you!

**Comment by dgehriger** [comment]
Update: 

I believe I understood that the solution involves enabling `MULTIMODULE`. I managed to compile the OpenI6X firmware using these settings and a few minor adjustments:

`PCB=I6X HELI=NO GVARS=NO LUA_COMPILER=NO MULTIMODULE=YES PCBI6X_ELRSV3=NO DEBUG=NO`

This fits just about into the firmware. So I guess my only question now is how the MULTIMODULE code should communicate with the the external RF module. I seem to have read that it's not possible to use the PPM output in software-serial mode. So which UART would you recommend I use?

**Reply by ajjjjjjjj** [reply]
@dgehriger reuse TX2 pad - same as CRSF.
